### ISSUE TITLE
zset member returned as an interface type instead of a string and is not type-safe

### ISSUE THREAD

**Issue by Aden-Q** [redis/go-redis#2817]
## Expected Behavior

According to the [redis doc](https://redis.io/docs/data-types/sorted-sets/), zset members should always be strings. We want zset members to be returned as strings so that we don't have to do type assertion explicitly.

## Current Behavior

zset member is returned as an interface: https://github.com/redis/go-redis/blob/21bd40a47e56e61c0598ea1bdf8e02e67d1aa651/sortedset_commands.go#L730

## Possible Solution

Define the Z struct's member field as a string type:
https://github.com/redis/go-redis/blob/21bd40a47e56e61c0598ea1bdf8e02e67d1aa651/sortedset_commands.go#L728-L731

**Comment by szuecs** [comment]
@chayim why this was a bug?
I would expect a breaking change only happens at major version updates.

**Comment by ofekshenawa** [comment]
Absolutely, I agree. We plan to release version 9.4.0 this Sunday, which will include the necessary fix.

**Comment by szuecs** [comment]
Thanks!

**Comment by ndyakov** [comment]
So, this is still a breaking change and is `Member` is still an `interface{}`. Keeping this alive for `v10`
